rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -562,15 +556,45 @@ impl PendingMessage {
                 0
             }
         };
+        // Skip this message if it has been retried too many times
+        if let Some(max_retries) = retries_before_skipping {
```

Comment:
feels a little awkward for this function to be the place where the decision of whether to skip is made?

I would've thought that we'd have `get_num_retries` always return a u32 that is accurate (easier to reason about it then imo), and then in `maybe_from_persisted_retries` we make the judgement call whether to ignore the message based off that number. The benefit being that this retry skipping logic isn't so intertwined in multiple places

Reply:
great point, I refactored so `get_num_retries` returns the true value but we can test both `get_num_retries` and the decision making at the same time